From: Luca Weiss <luca.weiss@fairphone.com>
Date: Fri, 10 Oct 2025 14:22:52 +0200
Subject: SQUASHME power: supply: qcom_smbx: fixups for smb5 introduction

---
 drivers/power/supply/qcom_smbx.c | 46 +++++++++++++++++++++++-----------------
 1 file changed, 27 insertions(+), 19 deletions(-)

diff --git a/drivers/power/supply/qcom_smbx.c b/drivers/power/supply/qcom_smbx.c
index d902f3f..e466eb0 100644
--- a/drivers/power/supply/qcom_smbx.c
+++ b/drivers/power/supply/qcom_smbx.c
@@ -196,9 +196,6 @@ enum smb_generation {
 #define CDP_CURRENT_UA					1500000
 #define DCP_CURRENT_UA					1500000
 #define CURRENT_MAX_UA					DCP_CURRENT_UA
-
-/* pmi8998 registers represent current in increments of 1/40th of an amp */
-#define CURRENT_SCALE_FACTOR				25000
 /* clang-format on */
 
 enum charger_status {
@@ -225,6 +222,9 @@ struct smb_init_register {
  * @base:		Base address for smb registers
  * @regmap:		Register map
  * @batt_info:		Battery data from DT
+ * @gen:		Generation of SMBx hardware block
+ * @current_step_size_ua: Step size of current limits in uA
+ * @current_limit_max_ua: Maximum charging current in uA
  * @status_change_work: Worker to handle plug/unplug events
  * @cable_irq:		USB plugin IRQ
  * @wakeup_enabled:	If the cable IRQ will cause a wakeup
@@ -239,6 +239,8 @@ struct smb_chip {
 	struct regmap *regmap;
 	struct power_supply_battery_info *batt_info;
 	enum smb_generation gen;
+	unsigned int current_step_size_ua;
+	unsigned int current_limit_max_ua;
 
 	struct delayed_work status_change_work;
 	int cable_irq;
@@ -254,7 +256,9 @@ struct smb_match_data {
 	const char *name;
 	enum smb_generation gen;
 	size_t init_seq_len;
-	const struct smb_init_register __counted_by(init_seq_len) *init_seq;
+	const struct smb_init_register *init_seq;
+	unsigned int current_step_size_ua;
+	unsigned int current_limit_max_ua;
 };
 
 static enum power_supply_property smb_properties[] = {
@@ -414,7 +418,7 @@ static inline int smb_get_current_limit(struct smb_chip *chip,
 	int rc = regmap_read(chip->regmap, chip->base + ICL_STATUS(chip), val);
 
 	if (rc >= 0)
-		*val *= CURRENT_SCALE_FACTOR;
+		*val *= chip->current_step_size_ua;
 	return rc;
 }
 
@@ -422,12 +426,12 @@ static int smb_set_current_limit(struct smb_chip *chip, unsigned int val)
 {
 	unsigned char val_raw;
 
-	if (val > 4800000) {
+	if (val > chip->current_limit_max_ua) {
 		dev_err(chip->dev,
-			"Can't set current limit higher than 4800000uA");
+			"Can't set current limit higher than %u uA\n", chip->current_limit_max_ua);
 		return -EINVAL;
 	}
-	val_raw = val / CURRENT_SCALE_FACTOR;
+	val_raw = val / chip->current_step_size_ua;
 
 	return regmap_write(chip->regmap, chip->base + USBIN_CURRENT_LIMIT_CFG,
 			    val_raw);
@@ -585,6 +589,8 @@ static int smb_get_prop_health(struct smb_chip *chip, int *val)
 	case SMB5:
 		return smb5_get_prop_health(chip, val);
 	}
+
+	return -EINVAL;
 }
 
 static int smb_get_property(struct power_supply *psy,
@@ -793,15 +799,6 @@ static const struct smb_init_register smb5_init_seq[] = {
 			| USBIN_AICL_EN_BIT | SUSPEND_ON_COLLAPSE_USBIN_BIT,
 	  .val = USBIN_AICL_PERIODIC_RERUN_EN_BIT | USBIN_AICL_ADC_EN_BIT
 			| USBIN_AICL_EN_BIT | SUSPEND_ON_COLLAPSE_USBIN_BIT },
-	/*
-	 * This overrides all of the other current limit configs and is
-	 * expected to be used for setting limits based on temperature.
-	 * We set some relatively safe default value while still allowing
-	 * a comfortably fast charging rate.
-	 */
-	{ .addr = FAST_CHARGE_CURRENT_CFG,
-	  .mask = FAST_CHARGE_CURRENT_SETTING_MASK,
-	  .val = 1950000 / CURRENT_SCALE_FACTOR },
 };
 
 /* Init sequence derived from vendor downstream driver */
@@ -895,7 +892,8 @@ static const struct smb_init_register smb2_init_seq[] = {
 	 */
 	{ .addr = PRE_CHARGE_CURRENT_CFG,
 	  .mask = PRE_CHARGE_CURRENT_SETTING_MASK,
-	  .val = 500000 / CURRENT_SCALE_FACTOR },
+	  /* 20 * step size 25000 uA = 500000 uA */
+	  .val = 20 },
 };
 
 struct smb_match_data pmi8998_match_data = {
@@ -903,6 +901,8 @@ struct smb_match_data pmi8998_match_data = {
 	.init_seq_len = ARRAY_SIZE(smb2_init_seq),
 	.name = "pmi8998",
 	.gen = SMB2,
+	.current_step_size_ua = 25000,
+	.current_limit_max_ua = 4800000,
 };
 
 struct smb_match_data pm660_match_data = {
@@ -910,6 +910,8 @@ struct smb_match_data pm660_match_data = {
 	.init_seq_len = ARRAY_SIZE(smb2_init_seq),
 	.name = "pm660",
 	.gen = SMB2,
+	.current_step_size_ua = 25000,
+	.current_limit_max_ua = 4800000,
 };
 
 struct smb_match_data pm8150b_match_data = {
@@ -917,6 +919,8 @@ struct smb_match_data pm8150b_match_data = {
 	.init_seq_len = ARRAY_SIZE(smb5_init_seq),
 	.name = "pm8150b",
 	.gen = SMB5,
+	.current_step_size_ua = 50000,
+	.current_limit_max_ua = 5000000,
 };
 
 struct smb_match_data pm7250b_match_data = {
@@ -924,6 +928,8 @@ struct smb_match_data pm7250b_match_data = {
 	.init_seq_len = ARRAY_SIZE(smb5_init_seq),
 	.name = "pm7250b",
 	.gen = SMB5,
+	.current_step_size_ua = 50000,
+	.current_limit_max_ua = 5000000,
 };
 
 
@@ -1008,6 +1014,8 @@ static int smb_probe(struct platform_device *pdev)
 	match_data = (const struct smb_match_data *)device_get_match_data(chip->dev);
 
 	chip->gen = match_data->gen;
+	chip->current_step_size_ua = match_data->current_step_size_ua;
+	chip->current_limit_max_ua = match_data->current_limit_max_ua;
 
 	dev_info(chip->dev, "Generation %s\n", chip->gen == SMB2 ? "SMB2" : "SMB5");
 
@@ -1087,7 +1095,7 @@ static int smb_probe(struct platform_device *pdev)
 	 * reporting.
 	 */
 	rc = regmap_write(chip->regmap, chip->base + FAST_CHARGE_CURRENT_CFG,
-			  1950000 / CURRENT_SCALE_FACTOR);
+			  1950000 / chip->current_step_size_ua);
 	if (rc < 0)
 		return dev_err_probe(chip->dev, rc,
 				     "Couldn't write fast charge current cfg");
