From: Stanislav Zaikin <zstaseg@gmail.com>
Date: Tue, 21 Oct 2025 13:54:51 +0200
Subject: ASoC: qcom: sm8250: Add Xiaomi Taoyao soundcard compatible

Signed-off-by: Stanislav Zaikin <zstaseg@gmail.com>
---
 sound/soc/qcom/sm8250.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/soc/qcom/sm8250.c b/sound/soc/qcom/sm8250.c
index ce5b0059..80b213c 100644
--- a/sound/soc/qcom/sm8250.c
+++ b/sound/soc/qcom/sm8250.c
@@ -213,6 +213,7 @@ static const struct of_device_id snd_sm8250_dt_match[] = {
 	{ .compatible = "qcom,qrb4210-rb2-sndcard", .data = "sm4250" },
 	{ .compatible = "qcom,qrb5165-rb5-sndcard", .data = "sm8250" },
 	{ .compatible = "qcom,sm8250-sndcard", .data = "sm8250" },
+	{ .compatible = "xiaomi,taoyao-sndcard", .data = "sm7325" },
 	{}
 };
 
