From: Caleb Connolly <caleb@connolly.tech>
Date: Mon, 13 Dec 2021 19:04:59 +0000
Subject: imx519: fixes for camss / v4l2

---
 drivers/media/i2c/imx519.c | 24 +++++++++++++++++++++++-
 1 file changed, 23 insertions(+), 1 deletion(-)

diff --git a/drivers/media/i2c/imx519.c b/drivers/media/i2c/imx519.c
index f818a3d..5d79920 100644
--- a/drivers/media/i2c/imx519.c
+++ b/drivers/media/i2c/imx519.c
@@ -35,7 +35,7 @@
 
 #define IMX519_XCLK_FREQ		24000000
 
-#define IMX519_DEFAULT_LINK_FREQ	408000000
+#define IMX519_DEFAULT_LINK_FREQ	493500000
 
 /* Pixel rate is fixed at 686MHz for all the modes */
 #define IMX519_PIXEL_RATE		426666667
@@ -972,6 +972,7 @@ struct imx519 {
 
 	struct v4l2_ctrl_handler ctrl_handler;
 	/* V4L2 Controls */
+	struct v4l2_ctrl *link_freq;
 	struct v4l2_ctrl *exposure;
 	struct v4l2_ctrl *vflip;
 	struct v4l2_ctrl *hflip;
@@ -1420,6 +1421,10 @@ static const struct v4l2_subdev_ops imx519_subdev_ops = {
 	.pad = &imx519_pad_ops,
 };
 
+static const struct media_entity_operations imx519_subdev_entity_ops = {
+	.link_validate = v4l2_subdev_link_validate,
+};
+
 /* Power/clock management functions */
 static int imx519_power_on(struct device *dev)
 {
@@ -1517,6 +1522,9 @@ static int imx519_init_controls(struct imx519 *imx519)
 	struct v4l2_fwnode_device_properties props;
 	unsigned int i;
 	int ret;
+	static const u64 link_freq[] = {
+		IMX519_DEFAULT_LINK_FREQ,
+	};
 
 	ctrl_hdlr = &imx519->ctrl_handler;
 	ret = v4l2_ctrl_handler_init(ctrl_hdlr, 16);
@@ -1528,6 +1536,13 @@ static int imx519_init_controls(struct imx519 *imx519)
 			  IMX519_PIXEL_RATE, IMX519_PIXEL_RATE, 1,
 			  IMX519_PIXEL_RATE);
 
+	imx519->link_freq = v4l2_ctrl_new_int_menu(ctrl_hdlr, NULL,
+					       V4L2_CID_LINK_FREQ,
+					       ARRAY_SIZE(link_freq) - 1,
+						   0, link_freq);
+	if (imx519->link_freq)
+		imx519->link_freq->flags |= V4L2_CTRL_FLAG_READ_ONLY;
+
 	/*
 	 * Create the controls here, but mode specific limits are setup
 	 * in the imx519_set_framing_limits() call below.
@@ -1704,6 +1719,12 @@ static int imx519_probe(struct i2c_client *client)
 		return dev_err_probe(dev, PTR_ERR(imx519->xclk),
 				     "failed to get xclk\n");
 
+	ret = clk_set_rate(imx519->xclk, IMX519_XCLK_FREQ);
+	if (ret < 0) {
+		dev_err(dev, "failed to set xclk rate\n");
+		return ret;
+	}
+
 	xclk_freq = clk_get_rate(imx519->xclk);
 	if (xclk_freq != IMX519_XCLK_FREQ)
 		return dev_err_probe(dev, -EINVAL,
@@ -1755,6 +1776,7 @@ static int imx519_probe(struct i2c_client *client)
 	/* Initialize subdev */
 	imx519->sd.flags |= V4L2_SUBDEV_FL_HAS_DEVNODE |
 			    V4L2_SUBDEV_FL_HAS_EVENTS;
+	imx519->sd.entity.ops = &imx519_subdev_entity_ops;
 	imx519->sd.entity.function = MEDIA_ENT_F_CAM_SENSOR;
 
 	/* Initialize source pads */
