From: Jens Reidel <adrian@travitia.xyz>
Date: Tue, 15 Apr 2025 21:47:22 +0200
Subject: ufs: qcom: Add some missing frequency to gear mappings

Signed-off-by: Jens Reidel <adrian@travitia.xyz>
---
 drivers/ufs/host/ufs-qcom.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/ufs/host/ufs-qcom.c b/drivers/ufs/host/ufs-qcom.c
index eba0e66..8600fb4 100644
--- a/drivers/ufs/host/ufs-qcom.c
+++ b/drivers/ufs/host/ufs-qcom.c
@@ -2184,7 +2184,9 @@ static u32 ufs_qcom_freq_to_gear_speed(struct ufs_hba *hba, unsigned long freq)
 	case 300000000:
 		gear = UFS_HS_G4;
 		break;
+	case 240000000:
 	case 201500000:
+	case 200000000:
 		gear = UFS_HS_G3;
 		break;
 	case 150000000:
@@ -2192,6 +2194,7 @@ static u32 ufs_qcom_freq_to_gear_speed(struct ufs_hba *hba, unsigned long freq)
 		gear = UFS_HS_G2;
 		break;
 	case 75000000:
+	case 50000000:
 	case 37500000:
 		gear = UFS_HS_G1;
 		break;
