From: Luca Weiss <luca.weiss@fairphone.com>
Date: Fri, 20 Dec 2024 11:27:51 +0100
Subject: power: supply: qcom_qg: sort includes

Signed-off-by: Luca Weiss <luca.weiss@fairphone.com>
---
 drivers/power/supply/qcom_qg.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/power/supply/qcom_qg.c b/drivers/power/supply/qcom_qg.c
index 8bd15e1..7a587b5 100644
--- a/drivers/power/supply/qcom_qg.c
+++ b/drivers/power/supply/qcom_qg.c
@@ -3,14 +3,14 @@
  * Copyright (c) 2024, Danila Tikhonov <danila@jiaxyga.com>
  */
 
-#include <linux/module.h>
-#include <linux/kernel.h>
+#include <linux/iio/consumer.h>
 #include <linux/init.h>
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/of.h>
 #include <linux/of_address.h>
 #include <linux/platform_device.h>
-#include <linux/of.h>
 #include <linux/power_supply.h>
-#include <linux/iio/consumer.h>
 #include <linux/regmap.h>
 
 /* SRAM */
