From: Luca Weiss <luca.weiss@fairphone.com>
Date: Fri, 20 Dec 2024 12:30:17 +0100
Subject: power: supply: qcom_qg: use nvmem framework for reading sdam

Signed-off-by: Luca Weiss <luca.weiss@fairphone.com>
---
 drivers/power/supply/qcom_qg.c | 23 ++++++++++++++---------
 1 file changed, 14 insertions(+), 9 deletions(-)

diff --git a/drivers/power/supply/qcom_qg.c b/drivers/power/supply/qcom_qg.c
index 4683cbe..0bd3d73 100644
--- a/drivers/power/supply/qcom_qg.c
+++ b/drivers/power/supply/qcom_qg.c
@@ -7,15 +7,13 @@
 #include <linux/init.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
+#include <linux/nvmem-consumer.h>
 #include <linux/of.h>
 #include <linux/of_address.h>
 #include <linux/platform_device.h>
 #include <linux/power_supply.h>
 #include <linux/regmap.h>
 
-/* SRAM */
-#define QG_SRAM_BASE	0xb600
-
 /* BATT offsets */
 #define QG_S2_NORMAL_AVG_V_DATA0_REG	0x80 /* 2-byte 0x80-0x81 */
 #define QG_S2_NORMAL_AVG_I_DATA0_REG	0x82 /* 2-byte 0x82-0x83 */
@@ -33,6 +31,8 @@ struct qcom_qg_chip {
 
 	struct iio_channel *batt_therm_chan;
 
+	struct nvmem_device *sdam;
+
 	struct power_supply *batt_psy;
 	struct power_supply_battery_info *batt_info;
 };
@@ -149,9 +149,8 @@ static int qcom_qg_get_property(struct power_supply *psy,
 			return ret;
 		break;
 	case POWER_SUPPLY_PROP_VOLTAGE_OCV:
-		ret = regmap_raw_read(chip->regmap,
-			QG_SRAM_BASE + QG_SDAM_OCV_OFFSET, &val->intval, 4);
-		if (ret)
+		ret = nvmem_device_read(chip->sdam, QG_SDAM_OCV_OFFSET, 4, &val->intval);
+		if (ret < 0)
 			return ret;
 		break;
 	case POWER_SUPPLY_PROP_CURRENT_NOW:
@@ -170,9 +169,9 @@ static int qcom_qg_get_property(struct power_supply *psy,
 		val->intval = chip->batt_info->charge_full_design_uah;
 		break;
 	case POWER_SUPPLY_PROP_CHARGE_FULL:
-		ret = regmap_raw_read(chip->regmap, QG_SRAM_BASE +
-				QG_SDAM_LEARNED_CAPACITY_OFFSET, &val->intval, 2);
-		if (ret)
+		ret = nvmem_device_read(chip->sdam,
+				QG_SDAM_LEARNED_CAPACITY_OFFSET, 2, &val->intval);
+		if (ret < 0)
 			return ret;
 		val->intval *= 1000; /* mAh to uAh */
 		break;
@@ -233,6 +232,12 @@ static int qcom_qg_probe(struct platform_device *pdev)
 		return dev_err_probe(chip->dev, PTR_ERR(chip->batt_therm_chan),
 				     "Couldn't get batt-therm IIO channel\n");
 
+	/* NVMEM for SDAM access */
+	chip->sdam = devm_nvmem_device_get(chip->dev, NULL);
+	if (IS_ERR(chip->sdam))
+		return dev_err_probe(chip->dev, PTR_ERR(chip->sdam),
+				     "Couldn't get SDAM nvmem device\n");
+
 	psy_cfg.drv_data = chip;
 	psy_cfg.of_node = chip->dev->of_node;
 
