From: Luca Weiss <luca.weiss@fairphone.com>
Date: Fri, 19 Sep 2025 13:21:04 +0200
Subject: power: supply: qcom_qg: 6.17 compile fix

---
 drivers/power/supply/qcom_qg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/supply/qcom_qg.c b/drivers/power/supply/qcom_qg.c
index 885664c..1827697 100644
--- a/drivers/power/supply/qcom_qg.c
+++ b/drivers/power/supply/qcom_qg.c
@@ -245,7 +245,7 @@ static int qcom_qg_probe(struct platform_device *pdev)
 				     "Couldn't get SDAM nvmem device\n");
 
 	psy_cfg.drv_data = chip;
-	psy_cfg.of_node = chip->dev->of_node;
+	psy_cfg.fwnode = dev_fwnode(chip->dev);
 
 	/* Power supply */
 	chip->batt_psy =
