From: Jens Reidel <adrian@mainlining.org>
Date: Mon, 22 Sep 2025 13:42:06 +0200
Subject: sound: soc: codecs: aw88261: Add support for OF

Signed-off-by: Jens Reidel <adrian@mainlining.org>
---
 sound/soc/codecs/aw88261.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/sound/soc/codecs/aw88261.c b/sound/soc/codecs/aw88261.c
index f3d37a3..c5d82e2 100644
--- a/sound/soc/codecs/aw88261.c
+++ b/sound/soc/codecs/aw88261.c
@@ -1279,6 +1279,13 @@ static int aw88261_i2c_probe(struct i2c_client *i2c)
 	return ret;
 }
 
+static const struct of_device_id aw88261_of_match[] = {
+	{ .compatible = "awinic,aw88261" },
+	{ .compatible = "awinic,aw88263s" },
+	{ }
+};
+MODULE_DEVICE_TABLE(of, aw88261_of_match);
+
 static const struct i2c_device_id aw88261_i2c_id[] = {
 	{ AW88261_I2C_NAME },
 	{ }
@@ -1288,6 +1295,7 @@ MODULE_DEVICE_TABLE(i2c, aw88261_i2c_id);
 static struct i2c_driver aw88261_i2c_driver = {
 	.driver = {
 		.name = AW88261_I2C_NAME,
+		.of_match_table = aw88261_of_match,
 	},
 	.probe = aw88261_i2c_probe,
 	.id_table = aw88261_i2c_id,
