From: Jens Reidel <adrian@mainlining.org>
Date: Sun, 21 Sep 2025 10:11:30 +0200
Subject: sound: soc-dai: Avoid a nullptr deref if driver has no ops set

Signed-off-by: Jens Reidel <adrian@mainlining.org>
---
 include/sound/soc-dai.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/sound/soc-dai.h b/include/sound/soc-dai.h
index 2243969..1b7bc8f 100644
--- a/include/sound/soc-dai.h
+++ b/include/sound/soc-dai.h
@@ -559,7 +559,7 @@ static inline void *snd_soc_dai_get_drvdata(struct snd_soc_dai *dai)
 static inline int snd_soc_dai_set_stream(struct snd_soc_dai *dai,
 					 void *stream, int direction)
 {
-	if (dai->driver->ops->set_stream)
+	if (dai->driver->ops && dai->driver->ops->set_stream)
 		return dai->driver->ops->set_stream(dai, stream, direction);
 	else
 		return -ENOTSUPP;
